OrientDBVisualizationTool/src/main/java/com/mycompany/orientdbvisualizationtool/controller/CollapseButtonsAction/CollapseButtonAction.java

```diff
@@ -0,0 +1,95 @@
+package com.mycompany.orientdbvisualizationtool.controller.CollapseButtonsAction;
+
+import com.mycompany.orientdbvisualizationtool.View.MainView;
+import javafx.scene.control.Button;
+import javafx.scene.control.SplitPane;
+import javafx.scene.layout.AnchorPane;
+
+/**
+ * Action taken when collapse button is clicked.
+ */
+public abstract class CollapseButtonAction {
+
+    private Button Collapse_Button;
+    private SplitPane Split_Pane;
+    private AnchorPane Collapse_Anchor_Pane;
+    private AnchorPane Center_Anchor_Pane;
+    private final int WIDTH = MainView.getWIDTH();
+    private boolean left;
+
+    /**
+     * constructor
+     * @param Collapse_Button button for collapsing pane
+     * @param Split_Pane pane containing the pane that is to be collapsed
+     * @param Collapse_Anchor_Pane pane to be collapsed/expanded
+     * @param Center_Anchor_Pane pane centered within split pane
+     * @param left for determining if the pane is left or right sided
+     */
+    public CollapseButtonAction(Button Collapse_Button, SplitPane Split_Pane, AnchorPane Collapse_Anchor_Pane, AnchorPane Center_Anchor_Pane, boolean left) {
+        this.Collapse_Button = Collapse_Button;
+        this.Split_Pane = Split_Pane;
+        this.Collapse_Anchor_Pane = Collapse_Anchor_Pane;
+        this.Center_Anchor_Pane = Center_Anchor_Pane;
+        this.left = left;
+        setPanelCollapseButtonProperty();
+    }
+
+    /**
+     * Sets up the properties of panel collapse or panel contract button
+     */
+    private void setPanelCollapseButtonProperty() {
+        if(left) {
```

Comment:
its better to refractor this huge if-else. also not a very good to so many if-else inside of a if-else

Reply:
noted.